core/unix/module_elf.h

```diff
@@ -110,6 +110,8 @@ typedef struct elf_loader_t {
 typedef byte *(*map_fn_t)(file_t f, size_t *size DR_PARAM_INOUT, uint64 offs, app_pc addr,
                           uint prot /*MEMPROT_*/, map_flags_t map_flags);
 typedef bool (*unmap_fn_t)(byte *map, size_t size);
+typedef byte *(*remap_fn_t)(file_t f, size_t *size DR_PARAM_INOUT, uint64 offs,
```

Comment:
Make it clear this cannot assume the existing mapping has the same file_t backing (to ensure no one tries to just use mremap).

In fact, it might be better to give this a different name, since "remap" has a close assocation with "mremap" and sounds like it just changes the size.  How about remap_as_file_nt_t or map_file_and_clobber_fn_t or map_file_onto_fn_t or map_file_on_anon_fn_t or something.

Reply:
It is not really a requirement that the prior map be an anonymous map, or that the new one be file-backed.

replace_map_fn_t seems better.

Reply:
Another option is replace_with_fixed_map_fn_t, but the "fixed" part seems like an implementation detail. The function is only supposed to provide the best way to replace an existing mapping with a new one.

Maybe overlap_map_fn_t? "overlap" communicates that there must be an existing mapping there, and also avoids the "re".

Reply:
SGTM